t/05-scheduler-full.t

```diff
@@ -122,16 +122,29 @@ subtest 're-scheduling and incompletion of jobs when worker rejects jobs or goes
     # simulate a worker in broken state; it will register itself but declare itself as broken
     my $broken_w_pid = broken_worker($api_key, $api_secret, "http://localhost:$mojoport", 3, 'out of order');
     wait_for_worker($schema, 5);
-    $allocated = scheduler_step();
+    # we do need the loop even after ensuring older workers
+    # are not considered with the `dead_workers` call in before as the just
+    # started worker can still be used as allocation target and needs a second
+    # cycle before the scheduler sees it as unusable
+    for (1 .. 2) {
+        $allocated = scheduler_step();
+        last if !$allocated || @$allocated == 0;
+        note "scheduler assigned to broken worker, waiting for unallocation, try: $_";
+    }
     is(@$allocated, 0, 'scheduler does not consider broken worker for allocating job');
```

Comment:
maybe a better alternative is that we check in `wait_for_worker` that the worker is in a steady "broken" state. Otherwise it could be the "broken worker" publishes itself as "idle" first before changing to "broken"

Reply:
excluded for now from this PR, trying in my branch fix/scheduler5_for_loops_as_well